librz/core/canalysis.c

```diff
@@ -7126,3 +7126,110 @@ RZ_API st64 rz_core_analysis_calls_count(RZ_NONNULL RzCore *core) {
 	}
 	return cov;
 }
+
+static const char *RzAnalysisNameTypeStrs[] = {
+	"var",
+	"function",
+	"flag",
+	"address",
+};
+
+/**
+ * \brief Convert \p typ to string (const char*)
+ */
+RZ_API RZ_BORROW const char *rz_analysis_name_type_to_str(RzAnalysisNameType typ) {
+	switch (typ) {
+	case RZ_ANALYSIS_NAME_TYPE_VAR:
+	case RZ_ANALYSIS_NAME_TYPE_FUNCTION:
+	case RZ_ANALYSIS_NAME_TYPE_FLAG:
+	case RZ_ANALYSIS_NAME_TYPE_ADDRESS:
+		return RzAnalysisNameTypeStrs[typ];
+	default:
+		rz_warn_if_reached();
+		return NULL;
+	}
+}
+
+RZ_API void rz_analysis_name_free(RZ_NULLABLE RzAnalysisName *p) {
+	if (!p) {
+		return;
+	}
+	free(p->name);
+	free(p->realname);
+	free(p);
+}
+
+/**
+ * \brief Rename whatever var/flag/function is used at core->offset to \p name
+ * \return success?
+ */
+RZ_API bool rz_core_analysis_rename(RZ_NONNULL RzCore *core, RZ_NONNULL const char *name) {
+	rz_return_val_if_fail(core && core->analysis && RZ_STR_ISNOTEMPTY(name), false);
+
+	RzAnalysisOp op;
+	ut64 tgt_addr = UT64_MAX;
+
+	rz_analysis_op(core->analysis, &op, core->offset,
+		core->block, 32, RZ_ANALYSIS_OP_MASK_BASIC);
+	RzAnalysisVar *var = rz_analysis_get_used_function_var(core->analysis, op.addr);
+	tgt_addr = op.jump != UT64_MAX ? op.jump : op.ptr;
+	rz_analysis_op_fini(&op);
+	if (var) {
+		return rz_analysis_var_rename(var, name, true);
+	} else if (tgt_addr != UT64_MAX) {
+		RzAnalysisFunction *fcn = rz_analysis_get_function_at(core->analysis, tgt_addr);
+		RzFlagItem *f = rz_flag_get_i(core->flags, tgt_addr);
+		if (fcn) {
+			return rz_analysis_function_rename(fcn, name);
+		} else if (f) {
+			return rz_flag_rename(core->flags, f, name);
+		} else {
+			return rz_flag_set(core->flags, name, tgt_addr, 1);
+		}
+	}
+	return false;
+}
+
+/**
+ * \brief Get information on whatever var/flag/function is used at core->offset
+ * \return RzAnalysisName
+ */
+RZ_API RZ_OWN RzAnalysisName *rz_core_analysis_name(RZ_NONNULL RzCore *core) {
```

Comment:
don't assume core->offset, pass an address as argument. This way the API can be more easily used even elsewhere by just passing a different address, without the need to switch core->offset value.